openwisp-monitoring/files/monitoring.agent

```diff
@@ -149,46 +176,36 @@ send_data() {
 			while true; do
 				if [ "$failures" -eq "$MAX_RETRIES" ]; then
 					[ -f "$RESPONSE_FILE" ] && error_message="\"$(cat "$RESPONSE_FILE")\"" || error_message='"".'
-					if [ "$VERBOSE_MODE" -eq "1" ]; then
-						logger -s "Data not sent successfully. Response code is \"$response_code\"." \
-							"Error message is $error_message" \
-							-p daemon.err
-					elif [ "$FAILING" -eq "0" ]; then
+					log -e -v "Data not sent successfully. Response code is \"$response_code\"." \
+						"Error message is $error_message"
+					# check if agent was already passing or not to avoid repeating log messages
+					if [ "$FAILING" -eq "0" ]; then
 						FAILING=1
-						logger -s "Data not sent successfully. Response code is \"$response_code\"." \
-							"Error message is $error_message" \
-							"Run with verbose mode to find more." \
-							-t openwisp-monitoring \
-							-p daemon.err
+						[ "$VERBOSE_MODE" -ne "1" ] && log -e -n "Data not sent successfully. Response code is \"$response_code\"." \
+							"Run with verbose mode to find more."
 					fi
 					break
 				fi
 				# send data
 				response_code=$($CURL_COMMAND -H "Content-Type: application/json" -d "$data" "$url")
 				if [ "$response_code" = "200" ]; then
-					if [ "$VERBOSE_MODE" -eq "1" ]; then
-						logger -s "Data sent successfully." \
-							-p daemon.info
-					elif [ "$FAILING" -eq "1" ]; then
-						logger -s "Data sent successfully." \
-							-t openwisp-monitoring \
-							-p daemon.info
+					log -i -v "Data sent successfully."
+					# check if agent was already failing or not to avoid repeating log messages
+					if [ "$FAILING" -eq "1" ]; then
 						FAILING=0
 						rm -f "$RESPONSE_FILE"
+						[ "$VERBOSE_MODE" -ne "1" ] && log -i -n "Data sent successfully."
```

Comment:
same here: `Data sent successfully.` is repeated twice and there's another unnecessary if here.

Reply:
I added this if so that we don't fill logs when data is transferred normally. In non-verbose mode, it will add this log "Data sent successfully" only once and the line will appear again if the connection is interrupted.
If the variable `FAILING` is set to 1, then only this line will be logged.